### PR TITLE
clean up shell variable expansion

### PR DESCRIPTION
**Pull request by cameronaaron**

### CODE REVIEW

File: src/server_manager/install_scripts/do_install_server.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env bash
```
Comment (trevj) [inline src/server_manager/install_scripts/do_install_server.sh]: Let's revert this, it's a little risky thanks to the multitude of environments this script has to run in.

File: src/server_manager/install_scripts/do_install_server.sh [hunk 2]
```diff
@@ -31,24 +32,24 @@ set -euxo pipefail
...
 export SHADOWBOX_DIR="${SHADOWBOX_DIR:-${HOME:-/root}/shadowbox}"
-mkdir -p $SHADOWBOX_DIR
+mkdir -p ${SHADOWBOX_DIR}
 
 # Save output for debugging
-exec 2>&1 >$SHADOWBOX_DIR/install-shadowbox-output
+exec 2>&1 >${SHADOWBOX_DIR}/install-shadowbox-output
 
 # Initialize sentry log file.
 export SENTRY_LOG_FILE="$SHADOWBOX_DIR/sentry-log-file.txt"
```
Comment (trevj) [inline src/server_manager/install_scripts/do_install_server.sh]: Needs braces?

File: src/server_manager/install_scripts/do_install_server.sh [hunk 3]
```diff
@@ -31,24 +32,24 @@ set -euxo pipefail
...
-> $SENTRY_LOG_FILE
+> ${SENTRY_LOG_FILE}
 function log_for_sentry() {
-  echo [$(date "+%Y-%m-%d@%H:%M:%S")] "do_install_server.sh" "$@" >>$SENTRY_LOG_FILE
+  echo [$(date "+%Y-%m-%d@%H:%M:%S")] "do_install_server.sh" "$@" >>${SENTRY_LOG_FILE}
 }
 function post_sentry_report() {
   if [[ -n "$SENTRY_API_URL" ]]; then
```
Comment (trevj) [inline src/server_manager/install_scripts/do_install_server.sh]: Braces for this check, do you think?

File: src/server_manager/install_scripts/do_install_server.sh [hunk 4]
```diff
@@ -31,24 +32,24 @@ set -euxo pipefail
...
     # Get JSON formatted string.  This command replaces newlines with literal '\n'
     # but otherwise assumes that there are no other characters to escape for JSON.
     # If we need better escaping, we can install the jq command line tool.
     readonly SENTRY_PAYLOAD_BYTE_LIMIT=8000
-    SENTRY_PAYLOAD="{\"message\": \"Install error:\n$(cat $SENTRY_LOG_FILE | awk '{printf "%s\\n", $0}' | tail --bytes $SENTRY_PAYLOAD_BYTE_LIMIT)\"}"
+    SENTRY_PAYLOAD="{\"message\": \"Install error:\n$(cat ${SENTRY_LOG_FILE} | awk '{printf "%s\\n", $0}' | tail --bytes ${SENTRY_PAYLOAD_BYTE_LIMIT})\"}"
     # See Sentry documentation at:
     # https://media.readthedocs.org/pdf/sentry/7.1.0/sentry.pdf
     curl "$SENTRY_API_URL" -H "Origin: shadowbox" --data-binary "$SENTRY_PAYLOAD"
```
Comment (trevj) [inline src/server_manager/install_scripts/do_install_server.sh]: Needs braces?

File: src/server_manager/install_scripts/do_install_server.sh [hunk 5]
```diff
@@ -136,13 +137,13 @@ export SB_PUBLIC_IP=$(cloud::public_ip)
...
-> $ACCESS_CONFIG
+> ${ACCESS_CONFIG}
 
 # Set trap which publishes an error tag and sentry report only if there is an error.
 function finish {
   INSTALL_SERVER_EXIT_CODE=$?
   log_for_sentry "In EXIT trap, exit code $INSTALL_SERVER_EXIT_CODE"
-  if [[ -z $(grep apiUrl $ACCESS_CONFIG) ]] || [[ -z $(grep certSha256 $ACCESS_CONFIG) ]]; then
+  if [[ -z $(grep apiUrl ${ACCESS_CONFIG}) ]] || [[ -z $(grep certSha256 ${ACCESS_CONFIG}) ]]; then
     echo "INSTALL_SCRIPT_FAILED: $INSTALL_SERVER_EXIT_CODE" | cloud::add_kv_tag "install-error"
```
Comment (trevj) [inline src/server_manager/install_scripts/do_install_server.sh]: Needs braces?

File: scripts/do_action.sh [hunk 6]
```diff
@@ -29,9 +29,11 @@ function do_action() {
   local action=$1
   echo -e "$STYLE_BOLD_WHITE[Running $action]$STYLE_RESET"
```
Comment (trevj) [inline scripts/do_action.sh]: Needs braces?

File: scripts/do_action.sh [hunk 7]
```diff
@@ -29,9 +29,11 @@ function do_action() {
...
   shift
-  $ROOT_DIR/src/${action}_action.sh "$@"
+  ${ROOT_DIR}/src/${action}_action.sh "$@"
   echo -e "$STYLE_BOLD_WHITE[Done $action]$STYLE_RESET"
```
Comment (trevj) [inline scripts/do_action.sh]: Needs braces?

File: scripts/do_action.sh [hunk 8]
```diff
@@ -29,9 +29,11 @@ function do_action() {
...
 }
 export -f do_action
 
 do_action "$@"
+
```
Comment (trevj) [inline scripts/do_action.sh]: Let's revert these two new lines.

File: src/server_manager/install_scripts/install_server.sh [hunk 9]
```diff
@@ -85,7 +86,7 @@ function command_exists {
 
 function log_for_sentry() {
   if [[ -n "$SENTRY_LOG_FILE" ]]; then
```
Comment (trevj) [inline src/server_manager/install_scripts/install_server.sh]: Needs braces?

File: src/server_manager/install_scripts/install_server.sh [hunk 10]
```diff
@@ -183,20 +184,20 @@ function remove_watchtower_container() {
...
 
 function remove_docker_container() {
-  $DOCKER_CMD rm -f $1
+  ${DOCKER_CMD} rm -f $1
 }
 
 function handle_docker_container_conflict() {
   local readonly CONTAINER_NAME=$1
   local readonly EXIT_ON_NEGATIVE_USER_RESPONSE=$2
   local PROMPT="> The container name \"$CONTAINER_NAME\" is already in use by another container. This may happen when running this script multiple times."
```
Comment (trevj) [inline src/server_manager/install_scripts/install_server.sh]: Needs braces?

File: src/server_manager/install_scripts/install_server.sh [hunk 11]
```diff
@@ -224,7 +225,7 @@ function get_random_port {
   until (( 1024 <= num && num < 65536)); do
     num=$(( $RANDOM + ($RANDOM % 2) * 32768 ));
```
Comment (trevj) [inline src/server_manager/install_scripts/install_server.sh]: Needs braces?

File: src/server_manager/install_scripts/install_server.sh [hunk 12]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env bash
```
Comment (trevj) [inline src/server_manager/install_scripts/install_server.sh]: Again, let's revert this - it probably is safe but I'd leave it for a separate PR.

**Comment by cameronaaron** [comment]
Thank you! I have made the changes! I am hoping they uploaded to Github correctly. I have been having some trouble with it today. Please let me know if anything more needs to be done

**Comment by trevj** [comment]
Thanks for the changes! Looks like you need to double-check that your changes have been uploaded, as in https://github.com/Jigsaw-Code/outline-server/pull/143.

**Comment by cameronaaron** [comment]
How does this look?

**Comment by trevj** [comment]
Closing for now - see https://github.com/Jigsaw-Code/outline-server/pull/205.
